server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java

```diff
@@ -80,33 +84,33 @@ class ScanDataSource implements DataSource {
     this.interruptFlag = interruptFlag;
     this.loadIters = loadIters;
     this.defaultLabels = tablet.getDefaultSecurityLabels();
+    this.scanDataSourceId = nextSourceId.incrementAndGet();
     if (log.isTraceEnabled()) {
```

Comment:
This doesn't even need the check if the trace is enabled. That's already built in to the logging API. We typically only need to do that when we need to do work to construct the message, and we want to avoid that work if we don't need to log the message. In this case, the only work we'd be avoiding is the construction of the varargs array for the format string's parameters, and that's assuming it's not hitting one of the optimized fixed-parameter overloaded versions of the log method from the logging library. Even if it was calling a varargs version, I doubt that would be measurable. We can just get rid of the isTraceEnabled check here.